external-docs/docs/guides/shape-data.md

```diff
@@ -1,3 +1,241 @@
 # Shaping your data
 
-todo: Explain the shapes of data we accept. Issue #17.
+> This page discusses the shapes of input data supported by csvcubed with the aim to help you understand which to use and how to generate data in the right shape.  
+
+csvcubed requires that all CSV data inputs are provided in one of two specialised form of [tidy data](../glossary/index.md#tidy-data):
+
+* the [canonical approach](#canonical-shape) - the **default recommended shape** accepted by csvcubed. It is the **most flexible** but also the most **verbose**.
+* the [pivoted approach](#pivoted-shape) - a **terser** shape currently only compatible with data sets containing a **single measure**.
+
+These two shapes share a number of similarities in how they require data to be structured; this is explored in the following section on the [common structure](#common-structure).
+
+## Common Structure
+
+Both data shapes expect that data is structured as per the following example:
+
+_Data set representing the number of 'Arthur's Bakes' stores in UK cities from 2020 to 2022_
+
+| Year | Location  | Value |      Status |
+|:-----|:----------|------:|------------:|
+| 2022 | London    |    35 | Provisional |
+| 2021 | Cardiff   |    26 |       Final |
+| 2020 | Edinburgh |    90 |       Final |
+| 2021 | Belfast   |     0 |       Final |
+
+**Note that**:
+
+1. The table has a flat and _tidy_ header row.
+    * No attempt has been made to group headers or bring any identifying characteristics such as the year into the headers.
+2. Each identifying characteristic has its own _dimension_ column.
+    * the `Year` and `Location` _dimension_ columns define the subset of the population that the row's observed value covers, i.e. each row describes the number of _Arthur's Bakes_ stores in a particular UK city in a given year.
+3. Every row has an _observed values_ column containing the measured value of some property.
+    * the `Value` column holds the observed values here.
+4. Each piece of information describing the _observed value_ has its own _attribute_ column.
+    * the `Status` _attribute_ column contains information describing the state of the observed value itself.
+    * Note that _attributes_ should only describe the observed value and must not identify any subset of the population.  
+
+## Canonical Shape
+
+The canonical shape extends the [common structure](#common-structure) by requiring that **each row** has a _measures_ column and a _units_ column; these columns define the measure and unit (of measure) for each row.
```

Comment:
Can we give an example here? E.g. "For example, the Measure (i.e. Number of 'Arthur's Bakes') and the Unit (i.e. Count)" in the first row tells that [what does it say]?